.github/workflows/check-whitespace.yml

```diff
@@ -13,38 +13,57 @@ jobs:
   check-whitespace:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
       with:
         fetch-depth: 0
 
     - name: git log --check
       id: check_out
       run: |
-        log=
+        problems=()
         commit=
-        while read dash etc
+        commitText=
+        lastcommit=
+        while read dash sha etc
         do
           case "${dash}" in
           "---")
-            commit="${etc}"
+            if test -z "${commit}"
```

Comment:
Was this meant to say `test -z "${lastcommit}"` instead?

Reply:
No.  `lastcommit` is really the good parent of the last error (guess I should rename (but it did start out as the last error)).  So, if `commit` is empty, it means an error was seen and 'lastcommit` needs to be reset for the parent.  Yes, it will be set on the first commit but that does not matter.

Reply:
Wait, what if the first commit in the `git log --check <range>` output already has issues? Then we set `lastcommit` to _that_ commit, and the `git rebase -i` hint is incorrect. Or am I missing something?

Reply:
In the `*` case, `commit` is set to empty so `lastcommit` is reset to `base.sha` at the end of the loop.  That is why setting for the first commit does not matter.